scripts/run_changed.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { resolve } = require('path');
+const { spawn } = require('child-process-promise');
+const simpleGit = require('simple-git/promise');
+const root = resolve(__dirname, '..');
+const git = simpleGit(root);
+
+/**
+ * Runs tests on packages changed (in comparison to origin/master...HEAD)
+ */
+
+async function runTestsOnChangedPackages() {
+  const diff = await git.diff(['--name-only', 'origin/master...HEAD']);
+  const changedFiles = diff.split('\n');
+  const changedPackages = {};
+  for (const filename of changedFiles) {
+    const match = filename.match('^(packages/[a-zA-Z0-9-]+)/.*');
+    if (match && match[1]) {
+      const pkg = require(resolve(root, match[1], 'package.json'));
+      if (pkg && pkg.scripts.test) {
+        changedPackages[match[1]] = true;
+      }
+    }
+  }
+  await runTests(Object.keys(changedPackages));
+}
+
+/**
+ * Runs `yarn test` in all dirs in pathList.
+ * @param {Array<string>} pathList
+ */
+async function runTests(pathList) {
+  if (!pathList) return;
```

Comment:
May be log this case: `No changes detected in any package`

Reply:
Good idea, added it to `runTestsOnChangedPackages` since this function is also used to run the integration tests.